plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -547,7 +552,22 @@ public synchronized void replaceTable(String databaseName, String tableName, Tab
 
     public synchronized void renameTable(String databaseName, String tableName, String newDatabaseName, String newTableName)
     {
-        setExclusive((delegate, hdfsEnvironment) -> delegate.renameTable(databaseName, tableName, newDatabaseName, newTableName));
+        setExclusive((delegate, hdfsEnvironment) -> {
+            Optional<Table> oldTable = delegate.getTable(databaseName, tableName);
+            boolean success = true;
+            try {
+                delegate.renameTable(databaseName, tableName, newDatabaseName, newTableName);
```

Comment:
if `oldTable.isEmpty()` is there anything to do?

Reply:
I would say that what needs to be done ( throw an exception / silently nod) is to be decided by the delegate.